src/EditorFeatures/Core.Wpf/InlineRename/UI/Adornment/RenameFlyout.xaml.cs

```diff
@@ -51,20 +54,26 @@ internal partial class RenameFlyout : InlineRenameAdornment
             _listener = listenerProvider.GetListener(FeatureAttribute.InlineRenameFlyout);
             _threadingContext = threadingContext;
             _wpfThemeService = themeService;
+            _identifierTextBox = new(() => new RenameUserInputTextBox(_viewModel));
+            _smartRenameUserInputComboBox = new(() => new SmartRenameUserInputComboBox(_viewModel));
 
             // On load focus the first tab target
             Loaded += (s, e) =>
             {
                 // Wait until load to position adornment for space negotiation
                 PositionAdornment();
 
-                IdentifierTextBox.Focus();
-                IdentifierTextBox.Select(_viewModel.StartingSelection.Start, _viewModel.StartingSelection.Length);
-                IdentifierTextBox.SelectionChanged += IdentifierTextBox_SelectionChanged;
+                RenameUserInput.Focus();
+                RenameUserInput.SelectText(_viewModel.StartingSelection.Start, _viewModel.StartingSelection.Length);
+                RenameUserInput.TextSelectionChanged += RenameUserInput_TextSelectionChanged;
+                RenameUserInput.GotFocus += RenameUserInput_GotFocus;
             };
 
             InitializeComponent();
 
+            RenameUserInputPresenter.Content = RenameUserInput;
+            RenameUserInput.PreviewKeyDown += RenameUserInput_PreviewKeyDown;
+
             // If smart rename is available, insert the control after the identifier text box.
             if (viewModel.SmartRenameViewModel is not null)
```

Comment:
It seems now SmartRenameControl is only responsible for display status text? (Because all the suggestions are displayed by the combo box). Can we rename it to something like `SmartRenameStatusControl`?